app/services/record_participant_event.rb

```diff
@@ -17,9 +17,10 @@ def required_params
 
   def call
     validate_schema!
-    return :not_found unless add_ect_profile_params
+    add_ect_profile_params!
     return :unprocessable_entity unless create_record
-    return :not_found unless valid_provider
+
+    valid_provider!
```

Comment:
Let's name it `validate_provider!` - it's no longer a method that returns a boolean, it's a method that sometimes raises an exception, so a verb over an adjective feels more fitting.